src/LinearOperators.jl

```diff
@@ -510,6 +537,93 @@ end
 
 opEye(nrow :: Int, ncol :: Int) = opEye(Float64, nrow, ncol)
 
+# TODO: not type stable
+"""
+    opEye!(X, Y, nrow, ncol)
+    opEye!(X, Y)
+
+Pre-allocated identity operator using `X` and `Y` as storage space. `Y` is used for
+transpose. `length(X) ≥ nrow` and `length(Y) ≥ ncol`.
+"""
+function opEye!(X :: Vector, Y :: Vector, nrow :: Int, ncol :: Int)
+  T = eltype(X)
+  @assert eltype(Y) == T
+  Xlen = length(X)
+  @assert Xlen ≥ nrow
```

Comment:
I think the lengths should be equal to simplify the code. Otherwise, you're returning copies of subvectors.